hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestValueQueue.java

```diff
@@ -106,14 +122,15 @@ public void testWarmUp() throws Exception {
   public void testRefill() throws Exception {
     MockFiller filler = new MockFiller();
     ValueQueue<String> vq =
-        new ValueQueue<String>(10, 0.1f, 300, 1,
+        new ValueQueue<String>(100, 0.1f, 30000, 1,
             SyncGenerationPolicy.ALL, filler);
+    // Trigger a prefill (10) and an async refill (91)
     Assert.assertEquals("test", vq.getNext("k1"));
-    Assert.assertEquals(1, filler.getTop().num);
-    // Trigger refill
-    vq.getNext("k1");
-    Assert.assertEquals(1, filler.getTop().num);
     Assert.assertEquals(10, filler.getTop().num);
+
+    // Wait for the async task to finish
+    waitForRefill(vq, "k1", 100);
+    Assert.assertEquals(91, filler.getTop().num);
```

Comment:
Can you explain in a comment where the number 91 comes from? If I understand correctly it's because the async replication task has to fill 91 values (`= 100 - 9`)